automation/infrastructure/terraform/kcs/install_kubeflow/monitoring.tf

```diff
@@ -0,0 +1,46 @@
+resource "null_resource" "install_monitoring" {
+  depends_on = [null_resource.now,
+    null_resource.helm_setup,
+    null_resource.prepare_prometheus,
+    null_resource.install_prometheus]
+}
+
+resource "null_resource" "helm_setup" {
+  count = var.kubeflow_monitoring == "prometheus" ? 1 : 0
+
+  provisioner "local-exec" {
+    command = <<EOT
+    helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+    helm repo update
+    EOT
+  }
+}
+
+resource "null_resource" "prepare_prometheus" {
+  triggers = {
+    kubeflow_monitoring            = var.kubeflow_monitoring
+    kubeconfig_filename             = var.kubeconfig_filename
+  }
+
+  count = var.kubeflow_monitoring == "prometheus" ? 1 : 0
+
+  provisioner "local-exec" {
+    command = "kubectl --kubeconfig=${self.triggers.kubeconfig_filename} apply -f monitoring.ns.yaml"
```

Comment:
technically you can call kubectl create as well instead of creating yaml and name can be taken as input.

Reply:
Done.